apps/labs/posts/ibis-an-idiomatic-flavor-of-sql-for-python-programmers.md

```diff
@@ -0,0 +1,1216 @@
+that has been in development for about 5 years; it currently has about
+1350 stars on Github. It provides an interface to SQL for Python
+programmers and bridges the gap between remote storage & execution
+systems. These features provide authors the ability to:
+
+1.  write backend-independent [SQL](https://en.wikipedia.org/wiki/SQL)
+    expressions in
+    [Python](https://en.wikipedia.org/wiki/Python_(programming_language));
+2.  access different database connections (eg.
+    [SQLite](https://www.sqlite.org/index.html),
+    [OmniSci](https://www.omnisci.com/),
+    [Pandas](http://pandas.pydata.org/)); and
+3.  confirm visually their SQL queries with [directed acyclic graphs
+    (DAGs)](https://en.wikipedia.org/wiki/Directed_acyclic_graph).
+
+Ibis is an alternative approach using databases that relies on Python
+rather than SQL experience. Typically, users have to learn an entirely
+new syntax or flavor of SQL to perform simple tasks. Now, those familiar
+with Python can avoid a new learning curve by using Ibis for composing
+and executing database queries using familiar Python syntaxes (i.e.,
+similar to Pandas and Dask). Ibis assists in formation of SQL
+expressions by providing visual feedback about each Python object. This
+post focuses on writing SQL expressions in Python and how to compose
+queries visually using Ibis. We'll demonstrate this with a SQLite
+database---in particular, [Sean Lahman's baseball
+database](http://www.seanlahman.com/baseball-archive/statistics/).
+
+## Connecting to a database
+
+To get started, we'll need to establish a [database
+connection](https://en.wikipedia.org/wiki/Database_connection). Ibis
+makes it easy to create connections of different types. Let's go ahead
+and do this now with the function
+[`ibis.sqlite.connect`](https://docs.ibis-project.org/docs/api.html#sqlite-client)
+(in this instance, the database used is a SQLite database):
+
+``` python
+%matplotlib inline
+import ibis
+import pathlib, requests
+
+db_path = pathlib.Path.cwd() / 'lahmansbaseballdb.sqlite'
+
+if not db_path.exists():          # Downloads database if necessary
+    with open(db_path, 'wb') as f:
+        URL = 'https://github.com/WebucatorTraining/lahman-baseball-mysql/raw/master/lahmansbaseballdb.sqlite'
+        req = requests.get(URL)
+        f.write(req.content)
+
+client = ibis.sqlite.connect(db_path.name) # Opens SQLite database connection
+```
+
+The `client` object represents our connection to the database. It is
+essential to use the appropriate Ibis connection---SQLite in this case
+constructed through the [`ibis.sqlite`
+namespace](https://docs.ibis-project.org/docs/api.html#sqlite-client)---for
+the particular database.
+
+This [baseball
+database](http://www.seanlahman.com/baseball-archive/statistics/) has 29
+distinct tables; we can see by running the following code:
+
+``` python
+tables = client.list_tables()
+print(f'This database has {len(tables)} tables.')
+```
+
+    This database has 29 tables.
```

Comment:
Same issue as described in:

- https://github.com/Quansight/Quansight-website/pull/647#pullrequestreview-1279809159